Mathlib/Tactic/Linarith/Lemmas.lean

```diff
@@ -24,51 +24,74 @@ namespace Linarith
 universe u
 theorem lt_irrefl {α : Type u} [Preorder α] {a : α} : ¬a < a := _root_.lt_irrefl a
 
-theorem zero_lt_one {α : Type u} [StrictOrderedSemiring α] : (0:α) < 1 := _root_.zero_lt_one
-
-theorem eq_of_eq_of_eq {α} [OrderedSemiring α] {a b : α} (ha : a = 0) (hb : b = 0) : a + b = 0 := by
+theorem eq_of_eq_of_eq {α} [Semiring α] {a b : α} (ha : a = 0) (hb : b = 0) : a + b = 0 := by
   simp [*]
 
-theorem le_of_eq_of_le {α} [OrderedSemiring α] {a b : α} (ha : a = 0) (hb : b ≤ 0) : a + b ≤ 0 := by
+section Semiring
+variable {α : Type u} [Semiring α] [PartialOrder α]
+
+theorem zero_lt_one [IsStrictOrderedRing α] : (0:α) < 1 :=
+  _root_.zero_lt_one
+
+theorem le_of_eq_of_le {a b : α} (ha : a = 0) (hb : b ≤ 0) : a + b ≤ 0 := by
   simp [*]
 
-theorem lt_of_eq_of_lt {α} [OrderedSemiring α] {a b : α} (ha : a = 0) (hb : b < 0) : a + b < 0 := by
+theorem lt_of_eq_of_lt {a b : α} (ha : a = 0) (hb : b < 0) : a + b < 0 := by
   simp [*]
 
-theorem le_of_le_of_eq {α} [OrderedSemiring α] {a b : α} (ha : a ≤ 0) (hb : b = 0) : a + b ≤ 0 := by
+theorem le_of_le_of_eq {a b : α} (ha : a ≤ 0) (hb : b = 0) : a + b ≤ 0 := by
   simp [*]
 
-theorem lt_of_lt_of_eq {α} [OrderedSemiring α] {a b : α} (ha : a < 0) (hb : b = 0) : a + b < 0 := by
+theorem lt_of_lt_of_eq {a b : α} (ha : a < 0) (hb : b = 0) : a + b < 0 := by
   simp [*]
 
-theorem add_nonpos {α : Type*} [OrderedSemiring α] {a b : α} (ha : a ≤ 0) (hb : b ≤ 0) :
+theorem add_nonpos [IsOrderedRing α] {a b : α} (ha : a ≤ 0) (hb : b ≤ 0) :
     a + b ≤ 0 :=
   _root_.add_nonpos ha hb
 
-theorem add_lt_of_le_of_neg {α : Type*} [StrictOrderedSemiring α] {a b c : α} (hbc : b ≤ c)
+theorem add_lt_of_le_of_neg [IsStrictOrderedRing α] {a b c : α} (hbc : b ≤ c)
     (ha : a < 0) : b + a < c :=
   _root_.add_lt_of_le_of_neg hbc ha
 
-theorem add_lt_of_neg_of_le {α : Type*} [StrictOrderedSemiring α] {a b c : α} (ha : a < 0)
+theorem add_lt_of_neg_of_le [IsStrictOrderedRing α] {a b c : α} (ha : a < 0)
     (hbc : b ≤ c) : a + b < c :=
   _root_.add_lt_of_neg_of_le ha hbc
 
-theorem add_neg {α : Type*} [StrictOrderedSemiring α] {a b : α} (ha : a < 0)
+theorem add_neg [IsStrictOrderedRing α] {a b : α} (ha : a < 0)
     (hb : b < 0) : a + b < 0 :=
   _root_.add_neg ha hb
 
-theorem mul_neg {α} [StrictOrderedRing α] {a b : α} (ha : a < 0) (hb : 0 < b) : b * a < 0 :=
+variable (α) in
+lemma natCast_nonneg [IsOrderedRing α] (n : ℕ) : (0 : α) ≤ n := Nat.cast_nonneg n
+
+@[deprecated (since := "2024-04-17")]
+alias nat_cast_nonneg := natCast_nonneg
```

Comment:
Where did this come from? This deprecation is a year old and can surely be deleted, but I can't even find it in the original version of the file.